### PR TITLE
refactor: add `fallback_to_local` region option

### PR DESCRIPTION
**Pull request by zyy17**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

Add `fallback_to_local` to control whether to fall back to local compaction if remote compaction fails.

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

File: src/mito2/src/compaction.rs [hunk 1]
```diff
@@ -314,6 +315,20 @@ impl CompactionScheduler {
...
+                            .compaction
+                            .fallback_to_local_compaction()
+                        {
+                            error!(e; "Failed to schedule remote compaction job for region {}", region_id);
+                            return Err(RemoteCompaction {
+                                region_id,
+                                job_id: None,
+                                reason: e.reason,
+                                location: Location::default(),
+                            });
```
Comment (evenyag) [inline src/mito2/src/compaction.rs:329]:
```suggestion
                            return RemoteCompactionSnafu {
                                region_id,
                                job_id: None,
                                reason: e.reason
                            }.fail();
```

**Review by evenyag — COMMENTED** [review]
You also need to add this key to
https://github.com/GreptimeTeam/greptimedb/blob/6fccff48100ceed4120cd59c401ffc66b4a77bf6/src/store-api/src/mito_engine_options.rs#L25-L26

**Comment by zyy17** [comment]
> You also need to add this key to
> 
> https://github.com/GreptimeTeam/greptimedb/blob/6fccff48100ceed4120cd59c401ffc66b4a77bf6/src/store-api/src/mito_engine_options.rs#L25-L26

oops。。。。I forget it again...

**Comment by zyy17** [comment]
@evenyag @v0y4g3r PTAL

File: src/mito2/src/region/options.rs [hunk 2]
```diff
@@ -201,6 +207,9 @@ pub struct TwcsOptions {
     /// Whether to use remote compaction.
     #[serde_as(as = "DisplayFromStr")]
     pub remote_compaction: bool,
+    /// Whether to fall back to local compaction if remote compaction fails.
+    #[serde_as(as = "DisplayFromStr")]
+    pub fallback_to_local_compaction: bool,
```
Comment (evenyag) [inline src/mito2/src/region/options.rs:212]: Maybe we can shorten this to `fallback_to_local`?

**Review by evenyag — APPROVED** [review]

**Review by WenyXu — APPROVED** [review]
